pkg/autoscaler/metrics/http_scrape_client.go

```diff
@@ -53,11 +58,27 @@ func (c *httpScrapeClient) Scrape(url string) (Stat, error) {
 	if resp.StatusCode < http.StatusOK || resp.StatusCode >= http.StatusMultipleChoices {
 		return emptyStat, fmt.Errorf("GET request for URL %q returned HTTP status %v", url, resp.StatusCode)
 	}
+	if resp.Header.Get("Content-Type") == network.ProtoAcceptContent {
+		return statFromProto(resp.Body)
+	}
+
+	return statFromPrometheus(resp.Body)
+}
 
-	return extractData(resp.Body)
+func statFromProto(body io.Reader) (Stat, error) {
+	var stat Stat
+	bodyBytes, err := ioutil.ReadAll(body)
```

Comment:
this seems like it could cause some unfortunate GC load in large envs where there are lots of stats being gathered since we'll be creating a temporary buffer in each of these ReadAlls (and we already saw quite a lot of mem usage in stats scraping in our profiles). I don't know if it's worth a sync.Pool here, but it might be. Makes me think it'd be worth adding a benchmark for this path?

Reply:
Ok sure I will have a look.

Reply:
@julz @markusthoemmes  here are the [bench-marking results](https://gist.github.com/skonto/61c8343976e1840d44d89cf9f798f2aa) for using a pool. It uses less bytes obviously per op, but number of allocations is only one less.

Reply:
I think we can get away with a much smaller buffer here. In my preliminary tests the whole messages was just 66 bytes or so. Maybe we should try a few real world examples and see how big it actually gets. Maybe 128 or 256 bytes is far and enough here?

Reply:
But as I said, imo non-blocking for this PR.

Reply:
(totally agree non-blocking for this PR)

Reply:
this is just an example i did try with 2048 at the beginning and since reducing it to 1024 didnt make much of a difference I just wanted to report the diff when using a pool. Yes a smaller buffer is the way to go.